### PR TITLE
Git chrisdias/reg2

### PR DESCRIPTION
**Pull request by chrisdias**
this is a big change to add registry support to the extension.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -398,17 +422,24 @@
...
   "dependencies": {
     "dockerfile-language-server-nodejs": "^0.0.7",
     "dockerode": "^2.5.1",
     "vscode-extension-telemetry": "^0.0.6",
-    "vscode-languageclient": "^3.1.0"
+    "vscode-languageclient": "^3.1.0",
+    "azure-arm-containerregistry": "^1.0.0-preview",
+    "azure-arm-resource": "^2.0.0-preview",
+    "azure-arm-website": "^1.0.0-preview",
+    "docker-hub-api": "^0.5.1",
```
Comment (rcjsuen) [inline package.json]: docker-hub-api is licensed under the [GPLv3](https://github.com/RyanTheAllmighty/Docker-Hub-API/blob/b00c8adf6bc1184a997263354762ea37533e2af1/LICENSE). I'm not a lawyer but you all might want to tread carefully here.

**Comment by chrisdias** [comment]
@rcjsuen thanks I removed the dependency.  I realize there is overlap with the `dockerHubApi.ts` module at the root of the repository, but I can fix that later.

File: dockerExtension.ts [hunk 2]
```diff
@@ -80,6 +88,17 @@ export function activate(ctx: vscode.ExtensionContext): void {
     ctx.subscriptions.push(vscode.commands.registerCommand('vscode-docker.compose.down', composeDown));
     ctx.subscriptions.push(vscode.commands.registerCommand('vscode-docker.system.prune', systemPrune));
 
+    ctx.subscriptions.push(vscode.commands.registerCommand('vscode-docker.createWebApp', async (context?: AzureImageNode | DockerHubImageNode) => {
+        const wizard = new WebAppCreator(outputChannel, azureAccount, context);
+        const result = await wizard.run();
+
+        if (result.status === 'Completed') {
```
Comment (lostintangent) [inline dockerExtension.ts]: Can we just delete this statement entirely?

File: explorer/deploy/wizard.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (lostintangent) [inline explorer/deploy/wizard.ts]: Is this and `webAppCreator.ts` largely a fork of the App Service extension?

File: dockerExtension.ts (hunk 2 above)
Comment (chrisdias) [reply dockerExtension.ts]: removed...

File: explorer/deploy/wizard.ts (hunk 3 above)
Comment (chrisdias) [reply explorer/deploy/wizard.ts]: yes, i'm hoping that we can consolidate all of these "creation" steps into a single package everyone can consume, but for now it is effectively a fork.

Comment (lostintangent) [reply explorer/deploy/wizard.ts]: Sounds good. I was just curious, so thanks for confirming. This change looks awesome!

**Pull request merged**